From 44b211bcf7ad58ff29e6495e1c3978e4660cb7d1 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 15 Jan 2019 18:04:34 -0500
Subject: [PATCH] OpenSSL: always provide OBJ_create() with name strings.

Some versions of OpenSSL seem to go back and forth as to whether NULL
for these names are okay.  Don't risk it.

Signed-off-by: Peter Jones <pjones@redhat.com>
Upstream-commit-id: 46b76a01717
---
 shim.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/shim.c b/shim.c
index a0eb19b..d7ee2b6 100644
--- a/shim.c
+++ b/shim.c
@@ -388,7 +388,9 @@ static BOOLEAN verify_eku(UINT8 *Cert, UINTN CertSize)
 	EXTENDED_KEY_USAGE *eku;
 	ASN1_OBJECT *module_signing;
 
-	module_signing = OBJ_nid2obj(OBJ_create(OID_EKU_MODSIGN, NULL, NULL));
+        module_signing = OBJ_nid2obj(OBJ_create(OID_EKU_MODSIGN,
+                                                "modsign-eku",
+                                                "modsign-eku"));
 
 	x509 = d2i_X509 (NULL, &Temp, (long) CertSize);
 	if (x509 != NULL) {
-- 
2.28.0

